### DISCUSSION TITLE
tower_sessions cookie not sent after oauth

### DISCUSSION THREAD

**Discussion by HosMercury** [maxcountryman/tower-sessions#197] [Q&A] [answered]
```
Ok(user) => {
    save_session_user(user.clone(), &session).await;
     json_data(json!(user)).into_response()
 }
```
                                    
```
pub async fn save_session_user(user: User, session: &Session) {
        session
            .insert(USER_SESSION_KEY, user)
            .await
            .expect("Saving user in session failed");
    }
```
    
    I do not know why the cookie was not sent after registering the user from OAuth data 
    
    while working from a regular user 
    
    I see the session added to db .. but the cookie was not sent after google oauth !! although this not related to oauth at all

**Comment by HosMercury** [comment]
It's weird and i don't know the reason

**Reply by maxcountryman** [reply]
It's hard to help without having a minimum example.

The most common reason a cookie isn't set is because the session is empty.

**Comment by HosMercury** [comment, marked as answer]
thx max
it was a trivial thing on Axios part .. I was not using ` withCredentials: true,`
I am writing that here, for maybe somebody who could face the same issue

**Reply by maxcountryman** [reply]
This is dupe of: https://github.com/maxcountryman/tower-sessions/discussions/193
